### PR TITLE
LiteDataLoader wrapper improvements

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #10279
Related: #10329

Adds small improvements around the init wrapper in Lite.
- Minimal docs
- Clearer variable names
- Separation of tests

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @borda @justusschock @awaelchli @akihironitta @carmocca

### CODE REVIEW

**Comment by carmocca** [comment]
@awaelchli do you want to land this before we move this code to Lightning?

File: tests/lite/test_lite.py [hunk 1]
```diff
@@ -444,25 +465,3 @@ def run(self):
...
-def test_replace_dataloader_init_method():
-    """Test that the context manager enables to save the parameters passed to the DataLoader __init__ method."""
-
-    class CustomDataLoader(DataLoader):
-        def __init__(self, extra_argument: int, *args, **kwargs):
-            super().__init__(*args, **kwargs)
-
-    dataloader = CustomDataLoader(extra_argument=1, dataset=range(1))
-    lite = EmptyLite()
-    with pytest.raises(MisconfigurationException, match="extra_argument"):
```
Comment (awaelchli) [inline tests/lite/test_lite.py:458]: reviewer: moved this part of the test to a separate test above `test_setup_dataloaders_raises_for_unknown_custom_args`,
the rest is already tested in `test_setup_dataloaders_with_custom_type` added in #10334

File: pytorch_lightning/lite/wrappers.py [hunk 2]
```diff
@@ -136,24 +140,17 @@ def recurse(cl: Type[Any]) -> None:
...
 
 
-def _enable_class(cls: Type[Any]) -> None:
-    cls._old_init = cls.__init__
-    cls.__init__ = _wrap_init(cls.__init__)
-
-
-def _disable_class(cls: Type[Any]) -> None:
-    cls.__init__ = cls._old_init
-    del cls._old_init
```
Comment (awaelchli) [inline pytorch_lightning/lite/wrappers.py:146]: reviewer: moved this into `_replace_dataloader_init_method` for clarity

File: pytorch_lightning/lite/wrappers.py [hunk 3]
```diff
@@ -110,21 +110,25 @@ def _convert_float_tensor(t: Tensor) -> Tensor:
...
+def _wrap_init(init: Callable) -> Callable:
+    """Wraps the ``__init__`` method of the dataloader in order to enable re-instantiation of custom subclasses of
+    :class:`~torch.utils.data.DataLoader`."""
+
+    @functools.wraps(init)
+    def wrapper(obj: DataLoader, *args: Any, **kwargs: Any) -> None:
+        params = dict(inspect.signature(obj.__init__).parameters)
         params.pop("args", None)
         params.pop("kwargs", None)
-        for init_name, init_arg in chain(zip(params, args), kwargs.items()):
```
Comment (awaelchli) [inline pytorch_lightning/lite/wrappers.py:119]: renamed variables for clarity

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by justusschock — APPROVED** [review]
